portal/serializers.py

```diff
@@ -0,0 +1,89 @@
+from rest_framework import serializers
+
+from portal.models import Poll, PollOption, PollVote
+
+
+class UserPollSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Poll
+        fields = ("id", "source", "question", "expire_date", "admin_comment")
+
+    def create(self, validated_data):
+        # adds user to the Poll
+        instance = Poll(**validated_data, user=self.context["request"].user)
+        # ensuring user cannot create an admin comment upon creation
+        instance.admin_comment = ""
+        instance.save()
+        return instance
+
+    def update(self, instance, validated_data):
+        user = self.context["request"].user
+        # checks if user is allowed to update or not
+        if user != instance.user:
```

Comment:
This should be done in a DRF permission

Reply:
instead of permissions, i just changed all of them via the get_queryset() function. basically, the queryset only contains the objects that they created